src/guards.ts

```diff
@@ -72,10 +97,17 @@ export function isBoolean<T>(data: T): data is Extract<T, boolean> {
  *    R.isPromise('somethingElse') //=> false
  * @category Guard
  */
-export function isPromise<T>(data: T): data is Extract<T, Promise<any>> {
+// @ts-expect-error
+export function isPromise<T>(data: T): data is DefinitelyPromise<T> {
```

Comment:
for some reason, the trick with making the parameter a union type (see e.g. `T | string` on _isString_) didn't work here for promises, I can't really pinpoint why, so this is the only ts-expect-error that is left.

Reply:
ok for me